### DISCUSSION TITLE
Trying encrypting json fields

### DISCUSSION THREAD

**Discussion by johnabil** [spatie/laravel-ciphersweet#24] [Q&A]
I have read the original documentation and the repo readme but I could not find anything that could help me through encrypting json fields but when I digged deep into the code of the package I found that when they encrypt the json field with json map configuration it takes the json column form the database without making json decode to the selected column and it known that by default that it should be decoded in Laravel so the package can access the json fields to decrypt it. 
so Can anyone help me with this issue??

**Comment by rabol** [comment]
Did you solve your issue?

**Reply by johnabil** [reply]
it's not solved but I already opened a pull request in paragonie/ciphersweet package with a description to the problem with the solution and I'm waiting their approval on the pull request [**fixing json field data with ciphersweet issue #82**](https://github.com/paragonie/ciphersweet/pull/82)

**Comment by rabol** [comment]
May I ask how you are storing the json data as I have problems storing

**Comment by johnabil** [comment]
@rabol I had a work around that you might use if you want

- a migration to encode the json column you need to be a json string column.
```php
ALTER TABLE table_name ALTER COLUMN json_column TYPE VARCHAR USING json_column #>> {}
``` 

- after that you will use ciphersweet to encode this field as a normal string field.
```php
public static function configureCipherSweet(EncryptedRow $encryptedRow): void
{
      $encryptedRow
          ->addField('json_column')
         ...
}
```
